pipeline/pipelines/core_generation_pipeline.py

```diff
@@ -55,13 +55,10 @@ class _GoCoreTaskFactory(code_gen.TaskFactoryBase):
     """
 
     def get_tasks(self, **kwargs):
-        return [
-            protoc_tasks.ProtoCodeGenTask('CoreProtoGen',
-                                          inject=kwargs),
-            protoc_tasks.GoExtractImportBaseTask('ExtractGoPackageName',
-                                                 inject=kwargs),
-            protoc_tasks.GoLangUpdateImportsTask('UpdateImports',
-                                                 inject=kwargs)]
+        tasks = [protoc_tasks.ProtoCodeGenTask,
+                 protoc_tasks.GoExtractImportBaseTask,
+                 protoc_tasks.GoLangUpdateImportsTask]
+        return task_utils.instantiate_tasks(tasks, kwargs)
```

Comment:
Another option is to use a recursive structure, where each element in a task list is either a task class or another task list, and the full tree could be instantiated before it needs to be run.